src/pl_bolts/models/self_supervised/swav/swav_swin.py

```diff
@@ -0,0 +1,910 @@
+                                    obj: an object to extract info from.
+            """
+            module = obj.__module__
+            if not module.startswith("torchvision"):
+                module = f"torchvision.internal.{module}"
+            name = obj.__class__.__name__
+            if isinstance(obj, FunctionType):
+                name = obj.__name__
+            torch._C._log_api_usage_once(f"{module}.{name}")
+
+    if not hasattr(torchvision.ops, "stochastic_depth"):
+
+        def stochastic_depth(input: Tensor, p: float, mode: str, training: bool = True) -> Tensor:
+            """Implements the Stochastic Depth from `"Deep Networks with Stochastic Depth"
+            <https://arxiv.org/abs/1603.09382>`_ used for randomly dropping residual branches
+            of residual architectures.
+
+            Args:
+                input: The input tensor or arbitrary dimensions with the first one
+                        being its batch i.e. a batch with ``N`` rows.
+                p: probability of the input to be zeroed.
+                mode: ``"batch"`` or ``"row"``.
+                    ``"batch"`` randomly zeroes the entire input, ``"row"`` zeroes randomly
+                    selected rows from the batch.
+                training: apply stochastic depth if is ``True``. Default: ``True``
+
+            Returns:
+                Tensor[N, ...]: The randomly zeroed tensor.
+            """
+            if not torch.jit.is_scripting() and not torch.jit.is_tracing():
+                _log_api_usage_once(stochastic_depth)
+            if p < 0.0 or p > 1.0:
+                raise ValueError(f"drop probability has to be between 0 and 1, but got {p}")
+            if mode not in ["batch", "row"]:
+                raise ValueError(f"mode has to be either 'batch' or 'row', but got {mode}")
+            if not training or p == 0.0:
+                return input
+
+            survival_rate = 1.0 - p
+            size = [input.shape[0]] + [1] * (input.ndim - 1) if mode == "row" else [1] * input.ndim
+            noise = torch.empty(size, dtype=input.dtype, device=input.device)
+            noise = noise.bernoulli_(survival_rate)
+            if survival_rate > 0.0:
+                noise.div_(survival_rate)
+            return input * noise
+
+        torch.fx.wrap("stochastic_depth")
+
+        class StochasticDepth(nn.Module):
+            """See :func:`stochastic_depth`."""
+
+            def __init__(self, p: float, mode: str) -> None:
+                super().__init__()
+                _log_api_usage_once(self)
+                self.p = p
+                self.mode = mode
+
+            def forward(self, input: Tensor) -> Tensor:
+                return stochastic_depth(input, self.p, self.mode, self.training)
+
+            def __repr__(self) -> str:
+                return f"{self.__class__.__name__}(p={self.p}, mode={self.mode})"
+
+    if not hasattr(torchvision.ops.misc, "MLP"):
+
+        class MLP(torch.nn.Sequential):
```

Comment:
let's rather create a wrapper on `init` which checks the compatibility, this declaration under if is tricky as a user may have it or not hard to guess if because of old Bolts or other dependency

Reply:
@Borda can you elaborate more and give me an short example of what you mean :sweat_smile:

Reply:
@Borda I have used the function `requires` you mentioned but I modified the if condition in line 13-16
by adding a not
` reqs = [
            ModuleAvailableCache(mod_ver) if "." **not** in mod_ver else RequirementCache(mod_ver)
            for mod_ver in module_path_version
        ]` 
As I think there was a bug in it, please check it out.